internal/elasticsearch/test/httptest.go

```diff
@@ -25,64 +20,34 @@ import (
 // elastic-package stack, and records the response.
 // Responses are recorded in the directory indicated by serverDataDir.
 func NewClient(t *testing.T, serverDataDir string) *elasticsearch.Client {
-	server := testElasticsearchServer(t, serverDataDir)
-	t.Cleanup(func() { server.Close() })
-
-	client, err := stack.NewElasticsearchClient(
-		elasticsearch.OptionWithAddress(server.URL),
-	)
-	require.NoError(t, err)
-
-	return client
-}
-
-func testElasticsearchServer(t *testing.T, mockServerDir string) *httptest.Server {
-	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		t.Log(r.Method, r.URL.String())
-		f := filepath.Join(mockServerDir, pathForURL(r.URL.String()))
-		if _, err := os.Stat(f); err != nil {
-			recordRequest(t, r, f)
-		}
-		http.ServeFile(w, r, f)
-	}))
-}
-
-var pathReplacer = strings.NewReplacer(
-	"/", "-",
-	"*", "_",
-	"?", "_",
-	"=", "_",
-)
-
-func pathForURL(url string) string {
-	clean := strings.Trim(url, "/")
-	if len(clean) == 0 {
-		return "root.json"
+	address := os.Getenv(stack.ElasticsearchHostEnv)
+	if address == "" {
+		address = "https://127.0.0.1:9200"
 	}
-	return pathReplacer.Replace(clean) + ".json"
-}
-
-func recordRequest(t *testing.T, r *http.Request, path string) {
-	client, err := stack.NewElasticsearchClient()
+	config, err := elasticsearch.NewConfig(
+		elasticsearch.OptionWithAddress(address),
+		elasticsearch.OptionWithPassword(os.Getenv(stack.ElasticsearchPasswordEnv)),
+		elasticsearch.OptionWithUsername(os.Getenv(stack.ElasticsearchUsernameEnv)),
+		elasticsearch.OptionWithCertificateAuthority(os.Getenv(stack.CACertificateEnv)),
+	)
 	require.NoError(t, err)
 
-	t.Logf("Recording %s in %s", r.URL.Path, path)
-	var recordURL url.URL
-	recordURL.Path = r.URL.Path
-	recordURL.RawQuery = r.URL.RawQuery
-
-	req, err := http.NewRequest(r.Method, recordURL.String(), nil)
+	rec, err := recorder.NewWithOptions(&recorder.Options{
+		CassetteName:       serverDataDir,
```

Comment:
Previously, all the responses were recorded in a directory, but currently all those responses are written in a single YAML file 

I'd rename this `serverDataDir` to indicate that now it's a file (cassette file). One option:
```suggestion
		CassetteName:       serverDataFile,
```

Same for the kibana/test/http_test.go

Reply:
Thanks, updated.